_posts/en/2019-01-20-rbf-in-the-wild.md

```diff
@@ -0,0 +1,242 @@
+
+### BRD: Generic Error Messages
+
+![BRD wallet transactions list screen showing a failed transaction screenshot](/img/posts/rbf-in-the-wild/rbf-brd-transaction-list-incoming-bumped.png)
+
+BRD shows a “Failed” label on the replacement transaction. In testing, the “Failed” transaction actually ends up being the transaction that confirms.
+
+## Usability Considerations
+
+A blanket recommendation for wallets, explorers, and other Bitcoin services would be to support Opt-in RBF,BIP125. Most of the services tested in this study do not have any support for RBF. As congestion builds on the network in the future, RBF will be an important tool in the scaling toolbox.
+
+Based on some of the examples above, there are some additional recommendations regarding RBF support.
+
+### Explorers Redirect to Replacement Transaction
+
+In almost all explorers, attempting to view the original transaction after the replacement confirms results in a 404 error. This can be confusing for users that have previously visited the original transaction's page on the explorer site and come back to find it has disappeared.
+
+Explorers should consider 301 redirects from the original to the replaced transaction that eventually confirms. Especially when the original transaction has been viewed by a visitor via that explorer. The redirection should be accompanied with an alert message indicating the original transaction has been replaced by current transaction.
+
+### Persist Original Transactions
+
+This recommendation is similar to previous one, only applied to wallets instead of explorers.
+
+Original transactions should be persisted in the UI, even subtly or under an advanced dialog. When sending a replacement transaction, a user may still want to have a reference to previously attempted, never-to-confirm transaction in their wallet.
+
+### Use RBF Compatible Block Explorers
+
+Wallets that support RBF and do not provide block explorer details within the app itself, should link to an RBF compatible block explorer. This will provide a consistent experience for the user as transactions will not “disappear” when transitioning from the RBF-supporting app to the non-RBF-supporting explorer.
+
+### Link to Explorer's Address Page
+
+Wallets/Exchanges should consider linking to the Explorer’s “address” page instead of “transaction” page. This can simplify the fact that transactions might be replaced/disappear from the explorer while also providing a single reliable link where someone can view the transaction.
+
+It is preferable that the explorer implement a redirect scheme similar to the one outlined above, however, this is a more user friendly alternative if that option is not available.
+
+### Label RBF Replaceable Transactions
+
+RBF signaled transactions could be visually labeled both in transaction list and transaction detail screens. Alternatively, RBF transactions could be the default and NON RBF transactions get an “unbumpable” type label.
```

Comment:
I'm not a fan of this, but I'm also not opposed to it.  I don't think users should need to know about whether or not a transaction opts in to replacement.  All they should need to know is whether or not their wallet supports a fee bumping feature.

Reply:
I added some language to clarify that knowing the replaceability is mostly valuable for knowing if the user can bump their own transaction.